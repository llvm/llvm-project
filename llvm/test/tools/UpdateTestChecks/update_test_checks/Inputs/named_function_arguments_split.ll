; Check that we split named function arguments correctly into a separate CHECK line,
; ensuring the opening parenthesis is on the label name, avoiding incorrect label
; matches if function names are not prefix free.
;
; RUN: opt < %s -passes=instsimplify -S | FileCheck %s
;
define i32 @"foo"(i32 %named) {
entry:
  ret i32 %named
}
