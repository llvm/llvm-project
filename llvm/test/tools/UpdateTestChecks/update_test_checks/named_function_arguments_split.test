# REQUIRES: x86-registered-target
## Basic test checking that update_test_checks.py works correctly
# RUN: cp -f %S/Inputs/named_function_arguments_split.ll %t.ll && %update_test_checks %t.ll --version=3
# RUN: diff -u %t.ll %S/Inputs/named_function_arguments_split.ll.expected
## Check that running the script again does not change the result:
# RUN: %update_test_checks %t.ll --version=3
# RUN: diff -u %t.ll %S/Inputs/named_function_arguments_split.ll.expected
