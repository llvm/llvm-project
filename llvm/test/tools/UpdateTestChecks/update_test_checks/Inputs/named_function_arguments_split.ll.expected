; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 3
; Check that we split named function arguments correctly into a separate CHECK line,
; ensuring the opening parenthesis is on the label name, avoiding incorrect label
; matches if function names are not prefix free.
;
; RUN: opt < %s -passes=instsimplify -S | FileCheck %s
;
define i32 @"foo"(i32 %named) {
; CHECK-LABEL: define i32 @foo(
; CHECK-SAME: i32 [[NAMED:%.*]]) {
; CHECK-NEXT:  entry:
; CHECK-NEXT:    ret i32 [[NAMED]]
;
entry:
  ret i32 %named
}
